csi/node_server.go

```diff
@@ -386,6 +386,28 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		return nil, err
 	}
 
+	// check if we need to resize the fs
+	// this is important since cloned volumes of bigger size don't trigger NodeExpandVolume
+	// therefore NodeExpandVolume is kind of redundant since we have to do this anyway
+	// some refs below for more details
+	// https://github.com/kubernetes/kubernetes/issues/94929
+	// https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/753
+	resizer := mount.NewResizeFs(utilexec.New())
```

Comment:
Does the CSI need to skip this check for `nfs`? Based on the ref links you provide, it seems that the util function does not handle `nfs`. 
And we don't need to do anything on the NFS client side for the expansion. Expanding the underlying filesystem of the NFS server work directory is enough, right?

Reply:
Correct, the share-manager is responsible for the filesystem expansion of the RWO volume that is exported via NFS.
NFS mount is handled at line 324 above this code path is only executed for regular RWO mount volumes. https://github.com/longhorn/longhorn-manager/pull/1064/files/3524e239cd7356171c93b7efe0f14d724b3ed16d..4fd4a3fbd27f9d622ac5a54b75cec31e1d3294dc#diff-e3208de2b41b1544cc85ea18e65fb55e847951129cdeee7db877af639c08e584R324

Reply:
Got it. I forgot to check the unchanged code hidden above.